cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/ScopeManager.kt

```diff
@@ -650,17 +658,41 @@ class ScopeManager : ScopeProvider {
             s =
                 if (scopes.isEmpty()) {
                     LOGGER.error(
-                        "Could not find the scope {} needed to resolve the call {}. Falling back to the current scope",
+                        "Could not find the scope {} needed to resolve the call {}. Falling back to the default (current) scope",
                         scopeName,
-                        call.name
+                        name
                     )
-                    currentScope
+                    scope
                 } else {
                     scopes[0]
                 }
         }
 
-        return resolve(s) { it.name.lastPartsMatch(call.name) && it.hasSignature(call.signature) }
+        return s
+    }
+
+    /**
+     * Directly jumps to a given scope.
+     *
+     * Handle with care, here be dragons. Should not be exposed outside of the cpg-core module.
```

Comment:
We should probably add "Returns the previous scope. Do not forget to set the scope back to the old scope after performing the actions inside this scope." or so to the documentation

Reply:
I made `jumpTo` private now.